cartography/intel/gcp/__init__.py

```diff
@@ -179,24 +172,81 @@ def start_gcp_ingestion(neo4j_session: neo4j.Session, config: Config) -> None:
         "UPDATE_TAG": config.update_tag,
     }
 
-    try:
-        crm_v1 = build_client("cloudresourcemanager", "v1")
-        crm_v2 = build_client("cloudresourcemanager", "v2")
-    except RuntimeError as e:
-        logger.warning(f"Unable to initialize GCP clients; skipping module: {e}")
-        return
-
-    # If we don't have perms to pull Orgs or Folders from GCP, we will skip safely
-    sync_gcp_organizations(
-        neo4j_session, crm_v1, config.update_tag, common_job_parameters
+    # IMPORTANT: We defer cleanup for hierarchical resources (orgs, folders, projects) and run them
+    # in reverse order. This prevents orphaned nodes when a parent is deleted.
+    # Without this, deleting an org would break its relationships to projects/folders, leaving them
+    # disconnected and unable to be cleaned up by their own cleanup jobs.
+    #
+    # Order of operations:
+    # 1. Sync all orgs
+    # 2. For each org:
+    #    a. Sync folders and projects
+    #    b. Sync project resources (with immediate cleanup)
+    #    c. Clean up projects and folders for this org
+    # 3. Clean up all orgs at the end
+    #
+    # This ensures children are cleaned up before their parents.
+
+    orgs = sync_gcp_organizations(
+        neo4j_session, config.update_tag, common_job_parameters, defer_cleanup=True
     )
-    sync_gcp_folders(neo4j_session, crm_v2, config.update_tag, common_job_parameters)
 
-    projects = get_gcp_projects(crm_v1)
+    # Track org cleanup jobs to run at the very end
+    org_cleanup_jobs = []
 
-    _sync_multiple_projects(
-        neo4j_session, projects, config.update_tag, common_job_parameters
-    )
+    # For each org, sync its folders and projects (as sub-resources), then ingest per-project services
+    for org in orgs:
+        org_resource_name = org.get("name", "")  # e.g., organizations/123456789012
+        if not org_resource_name or "/" not in org_resource_name:
+            logger.error(f"Invalid org resource name: {org_resource_name}")
+            continue
+
+        # Store the full resource name for cleanup operations
+        common_job_parameters["ORG_RESOURCE_NAME"] = org_resource_name
+
+        # Sync folders under org
+        folders = sync_gcp_folders(
+            neo4j_session,
+            config.update_tag,
+            common_job_parameters,
+            org_resource_name,
+            defer_cleanup=True,
```

Comment:
How about we make `sync_gcp_folders()` and `sync_gcp_projects()`not do cleanup at all? This way we don't need to add in the extra parameter and keep things to one codepath.

If the intent is to give users who import cartography as a library a choice, it seems like there's a decent amount of logic happening here in `_sync_project_resources` that they'd be missing out on, so they probably shouldn't be forking the functionality anyway.